File: 0782-jewels-and-stones/0782-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        jewel_box = {}
        output = 0
        for stone in stones:
            if stone in jewel_box:
                jewel_box[stone] += 1
            else:
                jewel_box[stone] = 1              
        for jewel in jewels:
            output += jewel_box.get(jewel,0)         
        return output

File: 0782-jewels-and-stones/0782-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashMap<Character, Integer> jewelBox = new HashMap<Character, Integer>();
        int output = 0;
        
        for(int i=0; i<stones.length(); i++) {
            if (jewelBox.containsKey(stones.charAt(i)))
                jewelBox.put(stones.charAt(i), jewelBox.get(stones.charAt(i))+1);
            else
                jewelBox.put(stones.charAt(i), 1);
        }
        
        for(int i=0; i<jewels.length();i++){
            if(jewelBox.get(jewels.charAt(i)) != null)
                output += jewelBox.get(jewels.charAt(i));
        }
        return output;
    }
}